### PR TITLE
Restrict acceptable values for Work access field

### LINKED ISSUE #169: Validate Work#access is in the set
* world
* stanford

We ought to consider using a Rails enum backed by a postgres enum. We've done this in other projects, AFAIK.

### PR DESCRIPTION
**Pull request by mjgiarlo**
Fixes #169 

## Why was this change made?

Because we want access values to be predictable and selected from a system-provided and -enforced field.

Includes:
* Switch DB schema to SQL
  * This allows us to use nice features like Postgres enums, which reinforce Rails-level validations with database-backed ones, because Rails and Ruby let everyone do everything.
* Restrict acceptable values for Work access field
  * This is implemented using a Rails enum backed by a PG enum. I have tested the migration in both directions both with an empty database and with some rows in it, and the migration works as expected.

## How was this change tested?

CI and with local migrations and rollbacks.

## Which documentation and/or configurations were updated?

None.

### CODE REVIEW

File: app/components/embargo_component.html.erb [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <div class="mb-3 row">
     <%= form.label :access, 'Who can access?', class: 'col-sm-2 col-form-label' %>
     <div class="col-sm-10">
-      <%= form.select :access, ['Stanford'], {}, class: "form-select" %>
+      <%= form.select :access, ['stanford'], {}, class: "form-select" %>
```
Comment (ndushay) [inline app/components/embargo_component.html.erb:6]: oooooooooo - nice catch.

File: app/models/work.rb [hunk 2]
```diff
@@ -9,16 +9,13 @@ class Work < ApplicationRecord
...
-  validates :citation, presence: true
-  validates :contact_email, presence: true
-  validates :created_edtf, presence: true
-  validates :license, presence: true
-  validates :state, presence: true
-  validates :subtype, presence: true
-  validates :title, presence: true
-  validates :work_type, presence: true
+  validates :abstract, :access, :citation, :contact_email, :created_edtf,
+            :license, :state, :subtype, :title, :work_type, presence: true
```
Comment (ndushay) [inline app/models/work.rb:13]: ya know, I wondered if I could do this ...  glad to know I could've.

File: spec/factories/works.rb [hunk 3]
```diff
@@ -10,7 +10,6 @@
     created_edtf { '1900' }
     abstract { 'test abstract' }
     citation { 'test citation' }
-    access { 'stanford' }
```
Comment (ndushay) [inline spec/factories/works.rb:13]: why does this go away?

**Review by ndushay — COMMENTED** [review]
LGTM, but I'm not confident enough in db foo to approve.   Still wondering who decided what the default should be (amy?  hannah?)

File: spec/factories/works.rb (hunk 3 above)
Comment (mjgiarlo) [reply spec/factories/works.rb:13]: I removed this so I could test that the DB (vs. this factory) was the actor responsible for setting the default value.

File: app/models/work.rb (hunk 2 above)
Comment (mjgiarlo) [reply app/models/work.rb:13]: :man_dancing:

File: app/components/embargo_component.html.erb (hunk 1 above)
Comment (mjgiarlo) [reply app/components/embargo_component.html.erb:6]: Tests caught it! :tada:

**Comment by mjgiarlo** [comment]
@ndushay :speech_balloon: 
> LGTM, but I'm not confident enough in db foo to approve. Still wondering who decided what the default should be (amy? hannah?)

I believe Amy gave the OK on Slack.

**Review by jcoyne — APPROVED** [review]

**Pull request merged**